packages/publisher/gcs/src/PublisherGCS.ts

```diff
@@ -0,0 +1,86 @@
+import PublisherBase, { PublisherOptions } from '@electron-forge/publisher-base';
+import { asyncOra } from '@electron-forge/async-ora';
+
+import debug from 'debug';
+import path from 'path';
+import { Storage } from '@google-cloud/storage';
+
+import { PublisherGCSConfig } from './Config';
+
+const d = debug('electron-forge:publish:gcs');
+
+type GCSArtifact = {
+  path: string;
+  keyPrefix: string;
+  platform: string;
+  arch: string;
+};
+
+export default class PublisherGCS extends PublisherBase<PublisherGCSConfig> {
+  name = 'gcs';
+
+  async publish({
+    makeResults,
+  }: PublisherOptions) {
+    const { config } = this;
+    const artifacts: GCSArtifact[] = [];
+
+    for (const makeResult of makeResults) {
+      artifacts.push(...makeResult.artifacts.map((artifact) => ({
+        path: artifact,
+        keyPrefix: config.folder || makeResult.packageJSON.version,
+        platform: makeResult.platform,
+        arch: makeResult.arch,
+      })));
+    }
+
+    const clientEmail = config.clientEmail || process.env.GOOGLE_CLOUD_CLIENT_EMAIL;
+    const privateKey = config.privateKey || process.env.GOOGLE_CLOUD_PRIVATE_KEY;
+
+    const storage = new Storage({
+      keyFilename: config.projectId || process.env.GOOGLE_APPLICATION_CREDENTIALS,
+      credentials: (clientEmail && privateKey) ? {
+        client_email: clientEmail,
+        private_key: privateKey,
+      } : undefined,
```

Comment:
Please extract the ternary into a regular if-then outside of the Storage constructor, it's hard to follow.